eZ/Bundle/EzPublishCoreBundle/Fragment/DirectFragmentRenderer.php

```diff
@@ -86,4 +97,38 @@ protected function getArguments(callable $controller, ControllerEvent $event): a
 
         return $arguments;
     }
+
+    /**
+     * @param string|\Symfony\Component\HttpKernel\Controller\ControllerReference $uri
+     *
+     * @throws \eZ\Publish\Core\MVC\Symfony\Templating\Exception\InvalidResponseException
+     */
+    public function render(
+        $uri,
+        Request $request,
+        array $options = []
+    ): Response {
+        $event = $this->getControllerEvent($request);
+        $controller = $this->getController($event);
+        $arguments = $this->getArguments($controller, $event);
+
+        $response = call_user_func_array($controller, $arguments);
```

Comment:
I'm pretty sure @alongosz would point this out, just as he did recently.

`call_user_func_array` is heavier than using calling a callable directly.

```suggestion
        $response = $controller(...$arguments);
```